### PR TITLE
Add unit tests for NodeMonitoringFailoverPlugin

### PR DESCRIPTION
**Pull request by karenc-bq**
### Summary

Unit tests for `NodeMonitoringFailoverPlugin.java`

### Additional Reviewers

@ColinKYuen 
@sergiyv-bitquill 
@brunos-bq

### CODE REVIEW

File: src/test/java/com/mysql/cj/jdbc/ha/ca/plugins/NodeMonitoringFailoverPluginTest.java [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+      final IFailoverPlugin failoverPlugin,
+      final Log log) {
+    Assertions.assertThrows(
+        NullArgumentException.class,
+        () -> plugin.init(set, info, failoverPlugin, log));
+  }
+
+  @Test
+  void test_2_initWithFailoverEnabled() {
+    Mockito
```
Comment (brunos-bq) [inline src/test/java/com/mysql/cj/jdbc/ha/ca/plugins/NodeMonitoringFailoverPluginTest.java:124]: It's better to have when/do/thenReturn static import, and not need to call Mockito.when().thenReturn() every time

File: src/test/java/com/mysql/cj/jdbc/ha/ca/plugins/NodeMonitoringFailoverPluginTest.java [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+  }
+
+  @Test
+  void test_4_executeWithFailoverDisabled() {
+    Mockito
+        .when(nativeFailureDetectionEnabledProperty.getValue())
+        .thenReturn(Boolean.FALSE);
+    initializePlugin();
+
+    Assertions.assertDoesNotThrow(() -> {
```
Comment (brunos-bq) [inline src/test/java/com/mysql/cj/jdbc/ha/ca/plugins/NodeMonitoringFailoverPluginTest.java:150]: Do we need assertDoesNotThrow? If you execute the code, and it doesn't throw an exception, that should be OK no?

File: src/test/java/com/mysql/cj/jdbc/ha/ca/plugins/NodeMonitoringFailoverPluginTest.java [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+      final Object result = plugin.execute(MONITOR_METHOD_NAME, sqlFunc);
+      Assertions.assertEquals(expected, result);
+    });
+
+    Mockito.verify(context, Mockito.atLeastOnce()).isNodeUnhealthy();
+    Mockito.verify(monitorService).stopMonitoring(Mockito.eq(context));
+    Mockito.verify(logger, Mockito.atLeastOnce()).logTrace(Mockito.anyString());
+  }
+
+  private static Stream<Arguments> generateNullArguments() {
```
Comment (brunos-bq) [inline src/test/java/com/mysql/cj/jdbc/ha/ca/plugins/NodeMonitoringFailoverPluginTest.java:254]: The code is easy to read, but i would suggest either adding a comment mentioning you are creating multiple objects with one single null argument

File: src/test/java/com/mysql/cj/jdbc/ha/ca/plugins/NodeMonitoringFailoverPluginTest.java (hunk 2 above)
Comment (karenc-bq) [reply src/test/java/com/mysql/cj/jdbc/ha/ca/plugins/NodeMonitoringFailoverPluginTest.java:150]: I think it makes the goal of the test more obvious, but I am okay with either approach.

File: src/test/java/com/mysql/cj/jdbc/ha/ca/plugins/NodeMonitoringFailoverPluginTest.java [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+
+    Assertions.assertThrows(Exception.class, () -> {
+      when(mockPlugin.execute(Mockito.any(), Mockito.any()))
+          .thenThrow(EXECUTION_EXCEPTION);
+
+      plugin.execute(MONITOR_METHOD_NAME, sqlFunc);
+    });
+
+    verify(monitorService).stopMonitoring(Mockito.eq(context));
+    verify(logger, Mockito.atLeastOnce()).logTrace(Mockito.anyString());
```
Comment (sergiyv-bitquill) [inline src/test/java/com/mysql/cj/jdbc/ha/ca/plugins/NodeMonitoringFailoverPluginTest.java:178]: I'm not sure we need to verify that. Logging isn't functional part here.

**Pull request merged**